### DISCUSSION TITLE
Extension developers are in a minority when voting is used to triage VS Code issues

### DISCUSSION THREAD

**Discussion by PEZ** [microsoft/vscode-discussions#287] [Extension Development QnA] [answered]
For VS Code issues, 20 thumbs-up votes are used as a threshold to automatically highlight issues for the VS Code team. This is very transparent and cool. However, as extension developers it can be quite hard for us to make 20 people understand the issue well enough to give it that thumb.

This is an example of an issue that got 28 votes to date. So I think it passed the triage step. But I am pretty sure that if VS Code users in general would understand what the issue would enable for them, it would get thousands of votes.

* https://github.com/microsoft/vscode/issues/135545

I think there should be some special triaging for issues with leverage, which Extension API issues often bring with them. Maybe it is a good first step to just make a special lane for issues about the API.

**Comment by ericdrobinson** [comment]
This is an excellent point.

VS Code **users** number in the millions, so asking for 20 thumbs up from the crowd of people who care enough to post issues is reasonable.

VS Code **developers** number in the thousands, so asking for 20 thumbs up from the crowd of people who encounter the same limit while developing an extension is going to be far, _far_ more rare.

Please reconsider the threshold for backlog admittance for API-focused issues.

**Comment by philliphoff** [comment]
Just so you don't feel alone, this has been a frustration of VS Code extension developers *within* the company, too. :-)

**Comment by worksofliam** [comment]
Hard agree. I also find it's those who spend the majority of their time using the API who ever have any requests, and usually those requests are quite detailed, which makes people less-inclined to read and agree to it (because it doesn't meet their need)

**Comment by isidorn** [comment]
To be honest, the number of votes does not always influence what will get tackled. Just because something gets to the backlog does not mean it will get worked on.
So instead of trying to get votes I think the best way to influence the VS Code team on what gets done is to provide clear benefits of adding that API while thinking how it aligns with the current VS Code API design. Just my 2 cents.

**Reply by ericdrobinson** [reply, marked as answer]
> To be honest, the number of votes does not always influence what will get tackled. Just because something gets to the backlog does not mean it will get worked on.

Yes, this is very much well understood. It is also beside the point. If an issue does **not** get sufficient votes, it is automatically **closed**. That is a problem as the default search that people do when looking for similar feature requests is:
<img width="252" alt="image" src="https://user-images.githubusercontent.com/9142587/207941128-4267518e-b6da-42f3-ae0d-67117d4d82c3.png">
So now if I'm going to look for a feature request regarding an API for extension development, then I have to remember to check for Closed issues as well. Perhaps this could be considered best practice but it is an extra step.

Additionally, if something was simply auto-closed then it gives the impression that your team didn't care about the request (even if they simply didn't _see_ it - I don't blame anyone for that as the main VS Code repository gets hammered with issues). There is no clear path to re-opening an issue and it is assumed that the VS Code team does **not** look at closed issues.

> So instead of trying to get votes I think the best way to influence the VS Code team on what gets done is to provide clear benefits of adding that API while thinking how it aligns with the current VS Code API design.

How do we ensure that the right people on the team actually _see_ the feature request before the bot auto-closes it? I would not expect VS Code developers to daily/weekly go through all the Issues and see if they can identify ones they may find relevant.

---

I personally don't really see a great path forward for this until either:

- We can specify (e.g. with a label?) that our feature request targets the VS Code Extension Development API and not a general VS Code end user feature, or
- The VS Code API is moved into a separate repository entirely wherein extension developers can have focused discussions, raise issues, request features about the API itself.

As things stand today, extension development API bugs/feature requests go into the same massive bucket as application bug/feature request reports. This results in the application of the same "rules" intended to manage issues from end users to those from the tiny percentage of end users who also develop extensions.
